### PR TITLE
Fix memory leaks for proc template

### PR DESCRIPTION
**Pull request by gassyfeve**
Fix https://github.com/sinatra/sinatra/issues/1714

### CODE REVIEW

File: lib/sinatra/base.rb [hunk 1]
```diff
@@ -892,17 +892,26 @@ def compile_template(engine, data, options, views)
...
+      when String
+        template_cache.fetch engine, data, options, views do
+          compile_block_template(template, options) { data }
+        end
+      else
+        raise ArgumentError, "Sorry, don't know how to render #{data.inspect}."
       end
     end
+
+    private
```
Comment (jkowens) [inline lib/sinatra/base.rb:907]: It looks like this method would already be marked private because of line 815.

**Review by jkowens — APPROVED** [review]
Looking good 👍

**Review by jkowens — CHANGES_REQUESTED** [review]
The only change I see would be to remove that private declaration.

**Comment by gassyfeve** [comment]
Already removed duplicate `private`

**Review by jkowens — APPROVED** [review]

**Comment by jkowens** [comment]
Thanks!

**Pull request merged**